sys/net/gnrc/network_layer/sixlowpan/nd/gnrc_sixlowpan_nd.c

```diff
@@ -181,7 +190,7 @@ kernel_pid_t gnrc_sixlowpan_nd_next_hop_l2addr(uint8_t *l2addr, uint8_t *l2addr_
         }
         return iface;
     }
-    else if ((nc_entry == NULL) || (!gnrc_ipv6_nc_is_reachable(nc_entry)) ||
+    if ((nc_entry == NULL) || (!gnrc_ipv6_nc_is_reachable(nc_entry)) ||
```

Comment:
could you move this check out of the enclosing if statement? Above l#166. Otherwise this will segfault, if `nc_entry` is still `NULL` after l#164 and if `next_hop` is not link local ..